### DISCUSSION TITLE
How do I just take the real part of the field from the function: output_efield_z()?

### DISCUSSION THREAD

**Discussion by 10UAikawa** [NanoComp/meep#2682] [Q&A]
If I have a complex field the function: output_efield_z() will output a complex field as given by it function in the source code.

def output_efield_z(sim):
    """
    Output the $z$ component of the field *e* (electric). If the field is complex, outputs
    two datasets, e.g. `ex.r` and `ex.i`, within the same HDF5 file for the real and
    imaginary parts, respectively.
    """
    sim.output_component(mp.Ez)

But if I want to create an animation, I need to only be using the real part of the field. So, using this function creates a problem when I want to create an animation using this code: 

sim.run(mp.at_beginning(mp.output_epsilon),
        mp.to_appended("ez", mp.at_every(0.6, mp.output_efield_z)),
        until=200)

Is there any way to get only the real part of the field?

**Comment by stevengj** [comment]
Why does it create a problem to have both `z.r` and `z.i` in the file when creating an animation?  What animation tool are you using?

**Reply by 10UAikawa** [reply]
I am using hdf5, and the problem is the dimensions. But I found a work around by removing the imaginary part in the .h5 file. I was also to remove certain dimensions in the .h5 file so I can view my 3D structure in 2D with the field overlayed.
